pkgs/desktops/deepin/update.nix

```diff
@@ -0,0 +1,37 @@
+{ lib, writeScript, coreutils, curl, gnugrep, gnused, jq, common-updater-scripts, nix }:
+{ name, ignored-versions ? "^2014\\.|^v[0-9]+" }:
+
+let
+  nameAndVersion = builtins.parseDrvName name;
+  packageVersion = nameAndVersion.version;
+  packageName = nameAndVersion.name;
+  attrPath = "deepin.${packageName}";
+in
+
+writeScript "update-${packageName}" ''
+  set -o errexit
+  set -x
+
+  # search for the latest version of the package on github
+  PATH=${lib.makeBinPath [ common-updater-scripts coreutils curl gnugrep jq ]}
```

Comment:
This will not be enough if user has GPG signing enabled.

Reply:
@jtojnar As I am not used to GPG signing, could you please be more specific on this? Do you mean that `PATH` is incomplete, and that something like `gnupg` should be added?

Currently the update script works for me.

Reply:
See https://help.github.com/articles/signing-commits/. At least `gnupg` is needed.

Reply:
It does not look like deepin are using gnupg. Is it used somewhere else at runtime?

Reply:
@Mic92 git requires it for signing.

Reply:
I suggest to also append the old PATH:

```bash
PATH=${lib.makeBinPath [ common-updater-scripts coreutils curl gnugrep jq ]}:$PATH
```

There might be other just as password asking agents if we start with gnupg.

Reply:
Actually, since the script does not run `git commit` directly, `gnupg` is not needed in its `PATH`. Sorry for the confusion.

Reply:
@Mic92 I would prefer not to append `$PATH`, as that includes the user current PATH setting. None of the update scripts I have found on the repository does that. It is better to list every needed binary path explicitly.

Reply:
@jtojnar I have removed `gnupg` then.